ThirdPartyAdapters/chartboost/chartboost/src/main/java/com/google/ads/mediation/chartboost/ChartboostSingleton.java

```diff
@@ -183,20 +191,43 @@ public static void startChartboostRewardedVideo(
         // Add this adapter delegate to mAdapterDelegates so that the events from
         // Chartboost SDK can be forwarded.
         addRewardedDelegate(location, adapterDelegate);
-        startChartboost((Activity) context, adapterDelegate.getChartboostParams(), adapterDelegate);
+        startChartboost(context,
+                adapterDelegate.getChartboostParams(),
+                adapterDelegate,
+                null);
+    }
+
+    /**
+     *  This method will initialize Chartboost SDK for banner ads and return whether or not
+     *  it successfully initialized.
+     * @param context
+     * @param adapterDelegate
+     * @param bannerListener
+     */
+    public static void startChartboostBanner(Context context,
+                                             AbstractChartboostAdapterDelegate adapterDelegate,
+                                             ChartboostBannerListener bannerListener) {
+        startChartboost(context,
+                adapterDelegate.getChartboostParams(),
+                adapterDelegate,
+                bannerListener);
+
+        initBannerWithListener(context, adapterDelegate.getChartboostParams(), bannerListener);
+        adapterDelegate.didInitialize();
```

Comment:
change the method to follow other pubs implementation which only creates one banner object per adapter.

Reply:
But by keeping the banner in the Chartboost Singleton, you are allowing for only one banner at the same time. Shouldn't we store the banner object in the adapter, so we can have as many banners as the publisher requests?

Reply:
I looked at the other publishers implementation and and sample adapters and all the banner are implement this way. I think we should follow as well since there is no strict documentation.

Reply:
Regardless of what other networks do, shouldn't we allow to use multiple banners?

Reply:
Adapter is build the way that it allows to return only one banner object anyway so I think multiple banners are not allow in admob. 
Take a looks at:
@Override
    public View getBannerView() {
        return ChartboostSingleton.mChartboostBanner;
    }